lms/djangoapps/courseware/views/index.py

```diff
@@ -193,6 +194,21 @@ def _redirect_if_needed_to_access_course(self):
         self._redirect_if_needed_to_register()
         self._redirect_if_needed_for_prereqs()
         self._redirect_if_needed_for_course_survey()
+        self._redirect_if_data_sharing_consent_needed()
+
+    def _redirect_if_data_sharing_consent_needed(self):
+        """
+        Determine if the user needs to provide data sharing consent before accessing
+        the course, and redirect the user to provide consent if needed.
+        """
+        course_id = unicode(self.course_key)
+        if consent_needed_for_course(self.real_user, course_id):
+            log.warning(
+                u'User %s cannot access the course %s because they have not granted consent',
+                self.real_user,
+                course_id,
+            )
+            raise Redirect(get_course_specific_consent_url(self.request, course_id))
```

Comment:
It's not how we do it, but it's what's being done in the other checkers on this class; I figured it was best to imitate those.